pharus/component_interface.py

```diff
@@ -4,13 +4,44 @@
 import datajoint as dj
 import re
 import inspect
-from datetime import datetime
+from datetime import date, datetime
 from flask import request, send_file
 from .interface import _DJConnector
 import os
 from pathlib import Path
 import types
 import io
+import numpy as np
+
+
+class NumpyEncoder(json.JSONEncoder):
+    """teach json to dump datetimes, etc"""
+
+    npmap = {
+        np.bool_: bool,
+        np.uint8: str,
+        np.uint16: str,
+        np.uint32: str,
+        np.uint64: str,
+        np.int8: str,
+        np.int16: str,
+        np.int32: str,
+        np.int64: str,
+        np.float32: str,
+        np.float64: str,
+        np.ndarray: list,
```

Comment:
This mapping doesn't seem right. Shouldn't it be more something like:

```suggestion
        np.bool_: bool,
        np.uint8: int,
        np.uint16: int,
        np.uint32: int,
        np.uint64: int,
        np.int8: int,
        np.int16: int,
        np.int32: int,
        np.int64: int,
        np.float32: float,
        np.float64: float,
        np.ndarray: list,
```

Reply:
The point is to convert it to json by overriding the base json encoder. Shouldn't it convert them to strings then? This is from IBL navigator.